src/Appwrite/Platform/Workers/UsageDump.php

```diff
@@ -83,109 +84,73 @@ public function action(Message $message, callable $getProjectDB): void
                         try {
                             $this->handleDatabaseStorage($key, $dbForProject);
                         } catch (\Exception $e) {
-                            console::error('[' . DateTime::now() . '] failed to calculate database storage for key [' . $key . '] ' . $e->getMessage());
+                            Console::error('[' . DateTime::now() . '] failed to calculate database storage for key [' . $key . '] ' . $e->getMessage());
                         }
                         continue;
                     }
 
+                    $documents = [];
+
                     foreach ($this->periods as $period => $format) {
```

Comment:
I would swap the order of the loops. 
First we loop over the keys in the stats array and for each key we iterate through the periods. Periods are fixed, `1h`, `1d` and `inf`

So in the current scenario if you have 1000 keys, you are going to make 1000 DB calls. the batching here is happening only for the periods. 

I would like us to swap the forloops, so that you are able to batch 1000 documents for each period separately and instead make 3 calls to the DB

Reply:
Couldn't swap the loops because $periods loop depends on $key from the outer loop, but I've moved the DB call to outside both loops, so there's only one write per project.

I've also added caching for the database/collection/collection size calls because they can be re-used within the same projects loop